packages/vue-urql/src/useQuery.ts

```diff
@@ -0,0 +1,177 @@
+import { inject, Ref, ref, watch, onMounted } from 'vue';
+import { DocumentNode } from 'graphql';
+import { pipe, subscribe, onEnd } from 'wonka';
+import {
+  CombinedError,
+  OperationContext,
+  RequestPolicy,
+  Operation,
+  Client,
+  createRequest,
+  GraphQLRequest,
+} from '@urql/core';
+import { onPush } from 'wonka';
+import { publish } from 'wonka';
+
+export interface UseQueryArgs<V> {
+  query: string | DocumentNode;
+  variables?: V;
+  requestPolicy?: RequestPolicy;
+  pollInterval?: number;
+  context?: Partial<OperationContext>;
+  pause?: boolean;
+}
+
+export interface UseQueryState<T> {
+  fetching: Ref<boolean>;
+  stale: Ref<boolean>;
+  data: Ref<T | undefined>;
+  error: Ref<CombinedError | undefined>;
+  extensions: Ref<Record<string, any> | undefined>;
+  operation: Ref<Operation | undefined>;
+  resume: () => void;
+  pause: () => void;
+  isPaused: Ref<boolean>;
+  executeQuery: () => void;
+}
+
+export type UseQueryResponse<T> = UseQueryState<T> & {
+  then: () => Promise<UseQueryState<T>>;
```

Comment:
Isn't there a `PromiseLike` built-in in TypeScript so that async/await and all other stuff works correctly?